File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraEditors;

namespace DXSample {

    public partial class Main : XtraForm {
        public Main() {
            InitializeComponent();
            Application.AddMessageFilter(new CustomMessageFilter(flyoutPanel1));
            flyoutPanel1.ButtonClick += OnButtonClick;
        }

        void OnButtonClick(object sender, FlyoutPanelButtonClickEventArgs e) {
            XtraMessageBox.Show(e.Button.Caption);
        }
        private void OnShowFlyoutPanelButtonClick(object sender, EventArgs e) {
            flyoutPanel1.ShowBeakForm(Control.MousePosition);
        }
        protected override void OnFormClosing(FormClosingEventArgs e) {
            flyoutPanel1.ButtonClick -= OnButtonClick;
            base.OnFormClosing(e);
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.Utils
Imports DevExpress.XtraEditors

Namespace DXSample

    Partial Public Class Main
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            Application.AddMessageFilter(New CustomMessageFilter(flyoutPanel1))
            AddHandler flyoutPanel1.ButtonClick, AddressOf OnButtonClick
        End Sub

        Private Sub OnButtonClick(ByVal sender As Object, ByVal e As FlyoutPanelButtonClickEventArgs)
            XtraMessageBox.Show(e.Button.Caption)
        End Sub
        Private Sub OnShowFlyoutPanelButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
            flyoutPanel1.ShowBeakForm(Control.MousePosition)
        End Sub
        Protected Overrides Sub OnFormClosing(ByVal e As FormClosingEventArgs)
            RemoveHandler flyoutPanel1.ButtonClick, AddressOf OnButtonClick
            MyBase.OnFormClosing(e)
        End Sub
    End Class
End Namespace